src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -177,5 +180,43 @@ protected static ITaskItem[] GenerateCommandLineArgsTaskItems(List<string> comma
 
             return items;
         }
+
+        private string GetToolDirectory()
+        {
+            var buildTask = typeof(ManagedToolTask).Assembly;
+            var buildTaskDirectory = Path.GetDirectoryName(buildTask.Location)!;
```

Comment:
If someone tries to load `Csc` in some context where Assembly.Location is empty (like wasm or ALC), perhaps we should throw an exception.